### DISCUSSION TITLE
Does apache pulsar support http protocol?

### DISCUSSION THREAD

**Discussion by preithig** [apache/pulsar#21497] [Q&A] [answered]
We do not want to use the pulsar protocol but http for our usecase. Does pulsar support it and is there any documentation on which version of http is supported?

**Comment by jak78** [comment]
Hello!
The [Netty Source Connector](https://docs.streamnative.io/hub/connector-netty-source-2.3#http) and the [HTTP Sink Connector](https://pulsar.apache.org/docs/3.1.x/io-http-sink/) allow you to publish and consume messages via HTTP, respectively. 
Could you please provide more details about your use case? It's important to note that using the HTTP protocol may not offer the same level of reliability and performance as the native protocol.

**Comment by preithig** [comment]
Hello!

Thank you for your response.
Our use case is for a hybrid solution where the requests are sent from a Cloud instance to the on-prem via Pulsar and the on-prem sends response back to the Cloud via Pulsar. 

Since most of the customers do not want to open a port in their firewall we are looking for a http solution. The solution you provided includes connector which is not feasible in customer's on-prem environment.

We tried the following where we are able to send http request via producer to Pulsar, but we were not able to consume it via http. We are using Pulsar 3.1.0 

https://github.com/apache/pulsar/wiki/PIP-64:-Introduce-REST-endpoints-for-producing,-consuming-and-reading-messages

Could you please let us know if there is a way we can achieve this?

**Reply by jak78** [reply]
Hello! Could you please tell us more about your issues with the REST endpoints?

**Reply by preithig** [reply]
We use the below to produce message - it is giving 200

POST /topics/(string: tenant_name)/(string: namespace_name)/(string: topic_name)

Sample request JSON
    {
        "messages": [
            {
                "payload": "{\"brand\":\"asus\",\"model\":\"rog\",\"year\":2020,\"gpu\":\"NVIDIA\",\"seller\":{\"state\":\"CA\",\"street\":\"back street\",\"zipCode\":90232}}"
            }
        ]
    }

Consume the messages from topic is giving 404.
 
1) Create a new consumer instance in the subscription 
POST /topics/(string: tenant_name)/(string: namespace_name)/(string: topic_name)/subscription/(string: subscription)
Payload: 
{
    "schema_type": "shared",
    "schema_version": 0
}
2) GET /topics/(string: tenant_name)/(string: namespace_name)/(string: topic_name)/subscription/(string: subscription)/consumer/(string: consumer_id)/messages


Both the topic and subscription exists, but the Create a new consumer instance in the subscription and Fetch messages from a given consumer is giving 404 for us

**Reply by jak78** [reply, marked as answer]
Hi!
This PIP is not implemented. Pulsar broker provides a [REST API](https://pulsar.apache.org/docs/3.1.x/client-libraries-rest/), but only message production is supported.
You could consider using [Websocket](https://pulsar.apache.org/docs/3.1.x/client-libraries-websocket/)

**Reply by preithig** [reply]
Thank you for your response. Websocket is not an option we could use.
